packages/react/src/runtime/ax.ts

```diff
@@ -27,8 +29,13 @@ const ATOMIC_GROUP_LENGTH = 5;
  *
  * @param classes
  */
-export default function ax(classNames: (string | undefined | null | false)[]): string | undefined {
-  if (classNames.length <= 1 && (!classNames[0] || classNames[0].indexOf(' ') === -1)) {
+export default function ax(
+  classNames: (string | undefined | null | false | AtomicGroups)[]
```

Comment:
With the introduction of `cx`, I believe this is the proper type as this does happen (or `cx` should change in a way I don't know how).

Realistically this could also be infinitely recursive, I don't think anything flattens here…but I won't solve a problem that doesn't really exist.